### ISSUE TITLE
Trying to log to Sentry.Serilog.SentrySink when other Serilog Sink fails

### ISSUE THREAD

**Issue by joeybrown** [getsentry/sentry-dotnet#335]
Please mark the type framework used:

- [ ] ASP.NET MVC
- [ ] ASP.NET Web API (OWIN)
- [x] ASP.NET Core
- [ ] WPF
- [ ] WinForms
- [ ] Xamarin
- [ ] Other:

Please mark the type of the runtime used:

- [ ] .NET Framework
- [ ] Mono
- [x] .NET Core

* Version: 3.0

Please mark the NuGet packages used:

- [ ] Sentry
- [x] Sentry.Serilog
- [ ] Sentry.NLog
- [ ] Sentry.Log4Net
- [ ] Sentry.Extensions.Logging
- [ ] Sentry.AspNetCore

* Version: 1.2.0

I would like to emit a `Serilog.Events.LogEvent` to an instance of `Sentry.Serilog.SentrySink` when there is an exception that occurs within a different Serilog Sink. In order to create an instance of `Sentry.Serilog.SentrySink` I must use reflection, because it is an `internal class`. I don't want to do that because I don't want to have to maintain that.

Here is an example of the code I'm using to create a `Sentry.Serilog.SentrySink` (following the pattern of `Serilog.SentrySinkExtensions.Sentry`):

```csharp
private static ILogEventSink GetSentrySink(Action<SentrySerilogOptions> configureOptions)
{
  var options = new SentrySerilogOptions();
  configureOptions?.Invoke(options);

  IDisposable sdkDisposable = null;
  if (options.InitializeSdk)
  {
    sdkDisposable = SentrySdk.Init(options);
  }

  var type = typeof(SentrySerilogOptions).Assembly.GetType("Sentry.Serilog.SentrySink");

  var constructor = type.GetConstructor(new[] {typeof(SentrySerilogOptions), typeof(IDisposable)});
  if (constructor == null)
  {
    throw new NullReferenceException("Cannot find SentrySink constructor");
  }

  var instance = constructor.Invoke(new object[] {options, sdkDisposable });
  return (ILogEventSink) instance;
}
```

Here is an example of why I use `SentrySink` (there's probably a better way of getting the timestamp from the message 😃):

```csharp
Serilog.Debugging.SelfLog.Enable(message =>
{
  try
  {
    const string regex = "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{7}Z";
    var match = Regex.Match(message, regex);
    var timestamp = DateTime.Parse(match.Value);
    var offset = new DateTimeOffset(timestamp);
    var exception = new Serilog.Debugging.LoggingFailedException(message);
    var messageTemplate = new MessageTemplate(message, new MessageTemplateToken[0]);
    var logEvent = new LogEvent(offset, LogEventLevel.Warning, exception, messageTemplate, new LogEventProperty[0]);
    sentrySink.Emit(logEvent);
  }
  catch
  {
    return;
  }
})
```

(FWIW, it is the Elasticsearch Sink from which I want to capture errors.)

I noticed that the `Serilog.Sinks.SystemConsole.ConsoleSink` is also marked `internal` which makes me wonder if there is a legitimate or idiomatic reason it should remain `internal`.

It seems like a simple change and I'm happy to do my best implementing the change.

I'm also open to further discussion.

Thank you for this excellent library.

**Comment by bruno-garcia** [comment]
We default to `internal` here unless we put some thinking into how to expose the API.
That doesn't mean we are not willing to open things up, we can always discuss it and change things.

That said, there might just be a better way to go about this, not that I'm aware. 
Perhaps Sentry integration for the Serilog Selflog? It should make sure not to capture SelfLogs from the Sentry Sink though.

We're happy to merge PRs btw, if you decide on this approach and only need a `public` Sentry sink.

**Comment by joeybrown** [comment]
Thank you so much for the prompt response.

I will not attempt to make this class `public` as I suggested. Like you suggested, I found a better way to go about this.

I am simply using the static `Serilog.Log` that is configured by from `Serilog.LoggerConfiguration.CreateLogger`.

Here is what my code now looks like:

```csharp
Serilog.Debugging.SelfLog.Enable(message =>
{
  var exception = new Serilog.Debugging.LoggingFailedException(message);
  Log.Error(exception, exception.Message);
});
```

I think this is a much cleaner approach than what I suggested, and it requires no code change on the sentry-dotnet side.

Thanks again!
